### DISCUSSION TITLE
RotatingFileSink minutely/flexible RotationPolicy

### DISCUSSION THREAD

**Discussion by lioriz1** [SpriteOvO/spdlog-rs#73] [Q&A]
Hi,
I'm using `RotatingFileSink` with `RotationPolicy::FileSize`, and I want to change the policy to time-dependent, and rotate every minute.

1. I saw that the [RotatingFileSink examples were removed](https://github.com/SpriteOvO/spdlog-rs/commit/6f9a095fb0688609d1da147e451c48591e02de23), was it changed?
2. Is there a way to rotate every minute, instead of every hour?

**Comment by SpriteOvO** [comment]
> I saw that the [RotatingFileSink examples were removed](https://github.com/SpriteOvO/spdlog-rs/commit/6f9a095fb0688609d1da147e451c48591e02de23), was it changed?

The API is not changed, examples for `FileSink` and `RotatingFileSink` are now in file https://github.com/SpriteOvO/spdlog-rs/blob/813edaf99d910387bb5c8413f8022919009ccc92/spdlog/examples/02_file.rs.

> Is there a way to rotate every minute, instead of every hour?

Currently it doesn't support rotating by minutes, but it shouldn't be too hard to implement. Before that, may I know what your real case is? Rotating by minutes means a lot of files will be generated. Rotating by hours and manually searching the logs for what you are interested in should be sufficient for most cases.

**Reply by lioriz1** [reply]
Thanks for your replay,
My current use case is that in my project there are big files that need to be rotated often.
But I guess that this should be flexible, like time logger, that can rotate every hour/minute/second

I will open an issue and PR, let me know what you think

**Reply by SpriteOvO** [reply]
I might be okay with a PR that adds a new rotation policy `Minutely`, as long as it is well tested and doesn't break any existing behavior.

But rotating by seconds doesn't make much sense to me, considering that "rotating" is a performance-expensive action, these will happen at that moment:

1. flushing the current file buffer, opening the new file and closing the old file;
2. renaming the file;
3. removing the old files.
4. ...

it just leads to poor performance and an unmanageable amount of log files.

@Lancern What do you think about this?

**Reply by SpriteOvO** [reply]
Oh, I think I get what you mean, a new rotating policy `Period(Duration)`, right? This sounds good to me as it can be very flexible, the users can set it to rotate every 6 hours, every 12 hours, every 30 minutes and so on.

I'd be happy to merge a PR in this way.

**Reply by Lancern** [reply]
> @Lancern What do you think about this?

The `Period(Duration)` idea sounds reasonable to me.

**Reply by lioriz** [reply]
@SpriteOvO @Lancern 
See PR: https://github.com/SpriteOvO/spdlog-rs/pull/74
